.github/workflows/windows-ci.yml

```diff
@@ -0,0 +1,288 @@
+name: windows-ci
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - windows-*.*.x
+    tags:
+      - windows-*
+
+  pull_request:
+    branches:
+      - '*'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  # INFO: We are cancelling the concurrency group if the change is on PR. For workflow dispatch, this will not work.
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+permissions:
+  id-token: write         # needed for AWS
```

Comment:
Could you introduce this in a separate PR?

Reply:
Yes, I can. I'm just testing for now, when everything is ok, I will create a new PR.